src/components/common/Header.tsx

```diff
@@ -0,0 +1,115 @@
+import styled from 'styled-components';
+import { Link, useNavigate } from 'react-router-dom';
+import { flexCenter } from 'styles/mixin';
+import { ReactComponent as CartIcon } from 'assets/cartIcon.svg';
+import { useAppSelector } from 'hooks/useAppSelector';
+import { useState } from 'react';
+import { useAppDispatch } from 'hooks/useAppDispatch';
+import { UserAction, UserActionType } from 'redux/actions/user';
+import theme from 'styles/theme';
+
+const Header = () => {
+  const navigate = useNavigate();
+  const { loading, error, data } = useAppSelector(state => state.userReducer);
```

Comment:
(마이너) 유저 정보는 많이 사용하다보니 훅으로 분리해주셔도 괜찮을 것 같아요
```suggestion
  const { userInfo, loading, error } = useUser();
```